lib/local_authorities_importer.rb

```diff
@@ -0,0 +1,63 @@
+require 'ostruct'
+
+class LocalAuthoritiesImporter
+  include JsonDownloader
+
+  LOCAL_AUTHORITY_TYPES = "DIS,LBO,MTD,UTA,COI,LGD"
+
+  def self.import_from_mapit
+    new.authorities_from_mapit
+  end
+
+  def authorities_from_mapit
+    url = "#{MAPIT_BASE_URL}areas/#{LOCAL_AUTHORITY_TYPES}"
+    mapit_las = get_mapit_authorities(url)
+
+    mapit_las.each do |mapit_la|
+      la = LocalAuthority.find_by_gss(mapit_la.gss)
+      if la
+        puts "Update to #{mapit_la.gss}"
+        Rails.logger.info("Updating authority #{mapit_la.name} (gss #{mapit_la.gss})")
+        la.update_attributes!(
+          name: mapit_la.name,
+          tier: mapit_la.tier)
+      else
+        puts "Creation of #{mapit_la.gss}"
+        Rails.logger.info("Creating authority #{mapit_la.name} (gss #{mapit_la.gss})")
+        LocalAuthority.create!(
+          gss: mapit_la.gss,
+          snac: mapit_la.snac,
+          name: mapit_la.name,
+          tier: mapit_la.tier)
+      end
+    end
+  end
+
+private
+
+  def get_mapit_authorities(url)
+    raw_authorities = get_json(url)
+    raw_authorities.values.map { |raw_authority|
+      get_la_struct(raw_authority)
+    }
+  end
+
+  def get_la_struct(raw_authority)
+    authority = OpenStruct.new
+    authority.name = raw_authority["name"]
+    authority.snac = raw_authority["codes"]["ons"]
+    authority.gss = raw_authority["codes"]["gss"]
+    authority.tier = identify_tier(raw_authority["type"])
+    authority
+  end
```

Comment:
We should avoid `OpenStruct` if we can (see: http://blog.honeybadger.io/how-openstruct-and-hashes-can-kill-performance/).  In this case we know what the attributes are, so we could define a `Struct` instead:

```
LAStruct = Struct.new(:name, :snac, :gss, :tier)
def get_la_struct(raw_authority)
  authority = LAStruct.new
  authority.name = raw_authority["name"]
  authority.snac = raw_authority["codes"]["ons"]
  authority.gss = raw_authority["codes"]["gss"]
  authority.tier = identify_tier(raw_authority["type"])
  authority
end
```

But! I'm not sure there's much benefit in having an object here at all when all we're going to do with it is immediately turn it into a new `LocalAuthority` or use it to update an existing one.  It's still useful to translate from the deeply nested hash we get from mapit though, but maybe a simple hash would suffice?